tests/test_transforms/test_inference.py

```diff
@@ -0,0 +1,1664 @@
+            ts, transform, train_segments=["segment_1", "segment_2"], expected_changes={}
+        )
+
+    @pytest.mark.parametrize(
+        "transform, dataset_name",
+        [
+            # decomposition
+            (
+                ChangePointsSegmentationTransform(
+                    in_column="target",
+                    change_point_model=RupturesChangePointsModel(change_point_model=Binseg(), n_bkps=5),
+                ),
+                "regular_ts",
+            ),
+            (
+                ChangePointsTrendTransform(
+                    in_column="target", change_point_model=Binseg(), detrend_model=LinearRegression(), n_bkps=5
+                ),
+                "regular_ts",
+            ),
+            (BinsegTrendTransform(in_column="target"), "regular_ts"),
+            (LinearTrendTransform(in_column="target"), "regular_ts"),
+            (TheilSenTrendTransform(in_column="target"), "regular_ts"),
+            (STLTransform(in_column="target", period=7), "regular_ts"),
+            (TrendTransform(in_column="target"), "regular_ts"),
+            # missing_values
+            (
+                ResampleWithDistributionTransform(
+                    in_column="regressor_exog", distribution_column="target", inplace=False, out_column="res"
+                ),
+                "ts_to_resample",
+            ),
+            (
+                ResampleWithDistributionTransform(
+                    in_column="regressor_exog", distribution_column="target", inplace=True
+                ),
+                "ts_to_resample",
+            ),
+            (
+                TimeSeriesImputerTransform(in_column="target"),
+                "ts_to_fill",
+            ),
+        ],
+    )
+    def test_transform_train_new_segments_failed_per_segment(self, transform, dataset_name, request):
+        ts = request.getfixturevalue(dataset_name)
+        with pytest.raises(NotImplementedError, match="Per-segment transforms can't work on new segments"):
+            self._test_transform_train_new_segments(
+                ts, transform, train_segments=["segment_1", "segment_2"], expected_changes={}
+            )
+
+    @to_be_fixed(raises=Exception)
+    @pytest.mark.parametrize(
+        "transform, dataset_name, expected_changes",
+        [
+            # encoders
+            (MeanSegmentEncoderTransform(), "regular_ts", {"create": {"segment_mean"}}),
+            (SegmentEncoderTransform(), "regular_ts", {"create": "segment_code"}),
+            # outliers
+            # TODO: error should be understandable, not like now
+            (DensityOutliersTransform(in_column="target"), "ts_with_outliers", {}),
+            (MedianOutliersTransform(in_column="target"), "ts_with_outliers", {}),
+            (PredictionIntervalOutliersTransform(in_column="target", model=ProphetModel), "ts_with_outliers", {}),
+            # math
+            # TODO: error should be understandable, not like now
```

Comment:
???

Reply:
It fails we very vague error and it is not really clear what happenned. I think may be we should raise some error that can be understood by the user.